### PR TITLE
Replace history import form

### PR DESCRIPTION
**Pull request by guerler**
The history import form is the last form which relies on makos. This PR replaces the form using js and redirects the post request to the existing api endpoint. It should be very easy to transform this new js-view into vue once available.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/histories.py [hunk 1]
```diff
@@ -317,9 +317,16 @@ def create(self, trans, payload, **kwd):
...
             archive_source = payload["archive_source"]
-            archive_type = payload.get("archive_type", "url")
+            archive_file = payload.get("archive_file")
+            if archive_source:
+                archive_type = payload.get("archive_type", "url")
+            elif hasattr(archive_file, "file"):
+                archive_source = payload["archive_file"].file.name
+                archive_type = "file"
+            else:
+                raise exceptions.MessageException("Please provide a url or file.")
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/histories.py:327]: Thanks for the backend fix - sorry about that issues!

Comment (guerler) [reply lib/galaxy/webapps/galaxy/api/histories.py:327]: No worries, thx for looking into it. It was helpful. I wonder if I can further improve the api parameter handling or if its good that archive_source is passed empty and then filled through the name in archive_file.

**Comment by mvdbeek** [comment]
@guerler I think I may have figured out your issue (https://github.com/galaxyproject/galaxy/issues/4345#issuecomment-346357704) with importing histories by URL (I had only tested file export/import, not the URL based import). If my changes are OK and this works for you i'm happy to merge.

**Review by mvdbeek — APPROVED** [review]

**Comment by jmchilton** [comment]
history-import.vue missing?

File: lib/galaxy/tools/imp_exp/imp_history_from_archive.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <tool id="__IMPORT_HISTORY__" name="Import History" version="0.1" tool_type="import_history">
   <type class="ImportHistoryTool" module="galaxy.tools"/>
   <action module="galaxy.tools.actions.history_imp_exp" class="ImportHistoryToolAction"/>
-  <command interpreter="python">unpack_tar_gz_archive.py "${ str( $__ARCHIVE_SOURCE__ ).encode( 'base64' ) }" "${ str( $__DEST_DIR__ ).encode( 'base64' ) }" --$__ARCHIVE_TYPE__ --encoded</command>
+  <command>python $__tool_directory__/unpack_tar_gz_archive.py '${ str( $__ARCHIVE_SOURCE__ ).encode( 'base64' ) }' '${ str( $__DEST_DIR__ ).encode( 'base64' ) }' --$__ARCHIVE_TYPE__ --encoded</command>
```
Comment (bgruening) [inline lib/galaxy/tools/imp_exp/imp_history_from_archive.xml]: `'$__tool_directory__/unpack_tar_gz_archive.py'`

Comment (guerler) [reply lib/galaxy/tools/imp_exp/imp_history_from_archive.xml]: I added quotes. Thanks @bgruening.

**Comment by guerler** [comment]
Thanks @mvdbeek for tracking the import failure issue. The export/import using the file and the webbrowser works. The url import still causes issues for me.

File: lib/galaxy/tools/imp_exp/imp_history_from_archive.xml (hunk 2 above)
Comment (mvdbeek) [reply lib/galaxy/tools/imp_exp/imp_history_from_archive.xml]: For the record, I don't think this is necessary and we don't do this in the tool XSD (https://docs.galaxyproject.org/en/master/dev/schema.html#id2).

Comment (mvdbeek) [reply lib/galaxy/tools/imp_exp/imp_history_from_archive.xml]: I'm wrong, definitely necessary and the schema has been updated in the dev branch.

**Comment by mvdbeek** [comment]
if you wget/curl the link, do you get the proper archive ?
I'm often forgetting to share the history, and then it fails with the same message because it's trying to unpack html ...

**Comment by guerler** [comment]
I am using `http://127.0.0.1:8080/history/export_archive?id=e49c4a047f65734a` and it returns a tar file which unpacks to cpgz

**Comment by guerler** [comment]
Thanks @mvdbeek that was the issue. It works fine now.

**Comment by mvdbeek** [comment]
Nice, glad this works!

**Pull request merged**